### PR TITLE
[WIP] : Adds method to list project langauges and data structure changes 🚧 

### PR DESCRIPTION
**Pull request by AdiAkhileshSingh15**
## Summary
This PR adds a function to list all the languages used in a project, iterating over the assets. Currently, it does not recognize all the languages properly due to limited languages mapped in `assets-config.js`, which we can add later anytime. (not sure if the languages that there are keeping in mind their extensive use in research, and no need for extension to it)

Discussion around addition of scanResult object, within or separate from, checklists data needed!

### Reviewer(s)
@lrasmus

### CODE REVIEW

**Comment by lrasmus** [comment]
Thank you @AdiAkhileshSingh15, reviewing now!

File: app/components/ReproChecklists/ReproChecklists.js [hunk 1]
```diff
@@ -82,41 +79,51 @@ const dummyChecklistItems = [
...
-
-      const findImageAssets = (asset) => {
-        if (asset.type === 'image') {
-          imageAssets.push(asset);
-        }
-        if (asset.children) {
-          asset.children.forEach(findImageAssets);
+function findAssetLanguagesAndDependencies(asset) {
+  if (asset.contentTypes.includes('code') && asset.type === 'file') {
+    const lastSep = asset.uri.lastIndexOf(path.sep);
```
Comment (lrasmus) [inline app/components/ReproChecklists/ReproChecklists.js:101]: Nothing wrong with this code, but we do have a util method (`AssetUtil.getExtensionFromUri`) that might work here?  Could you see and let me know if it doesn't do what you need?  Sorry too - I know there are a TON of utility methods.  No expectation for you to have known this existed.

https://github.com/StatTag/StatWrap/blob/repro-checklists/app/utils/asset.js#L350

**Review by lrasmus — APPROVED** [review]
@AdiAkhileshSingh15 - this looks good!  I flagged one spot I'm wondering if an existing utility method could be used, but I'm going to go ahead and merge this.

File: app/components/ReproChecklists/ReproChecklists.js (hunk 1 above)
Comment (AdiAkhileshSingh15) [reply app/components/ReproChecklists/ReproChecklists.js:101]: oh, didn't know about it. Definitely gonna replace with it!

**Pull request merged**